File: API/Models/ProjectModel.cs
using System.ComponentModel.DataAnnotations;
using API.DTO;

namespace API.Models
{
#nullable enable

    public class ProjectModel
    {
        [Key]
        public int ProjectId { get; set; }
        [Required]
        public string ProjectName { get; set; } = string.Empty;
        [Required]
        public string ProjectSn { get; set; } = string.Empty;
        [Required]
        public string ProjectDescription { get; set; } = string.Empty;
        public uint? DivisionId { get; set; }
        public DateTime? EstimatedStartDate { get; set; }
        public DateTime? EstimatedEndDate { get; set; }
        public DateTime? ActualStartDate { get; set; }
        public DateTime? ActualEndDate { get; set; }
        public uint? ProjectManagerId { get; set; }
        public decimal? EstimatedTotalHours { get; set; }
        public decimal? ActualTotalHours { get; set; }
        public decimal? EstimatedLaborCost { get; set; }
        public decimal? ActualLaborCost { get; set; }
        public decimal? EstimatedMaterialCost { get; set; }
        public decimal? ActualMaterialCost { get; set; }
        public ProjectStatusModel ProjectStatus { get; set; }
        public ICollection<TaskModel> Task { get; set; }
        public ICollection<ResourceModel> Resource { get; set; }
        public ClientModel Client { get; set; }


        public static implicit operator ProjectDto(ProjectModel project)
        {
            return new ProjectDto
            {
                ProjectId = project.ProjectId,
                ProjectSn = project.ProjectSn,
                ProjectName = project.ProjectName,
                ProjectDescription = project.ProjectDescription,
                ClientId = project.Client.ClientSn,
                DivisionId = project.DivisionId,
                EstimatedStartDate = project.EstimatedStartDate,
                EstimatedEndDate = project.EstimatedEndDate,
                ActualStartDate = project.ActualStartDate,
                ActualEndDate = project.ActualEndDate,
                ProjectManagerId = project.ProjectManagerId,
                EstimatedTotalHours = project.EstimatedTotalHours,
                ActualTotalHours = project.ActualTotalHours,
                ActualLaborCost = project.ActualLaborCost,
                EstimatedMaterialCost = project.EstimatedMaterialCost,
                ActualMaterialCost = project.EstimatedMaterialCost,
                ProjectStatusId = project.ProjectStatus.Id
            };
        }
    }
}

File: Client/src/models/project-model.ts
export class ProjectModel {

  public projectSn: string;
  public projectName: string;
  public projectDescription: string;
  public clientId: number;
  public divisionId: number;
  public estimatedStartDate: Date;
  public estimatedEndDate: Date;
  public actualStartDate?: Date;
  public actualEndDate?: Date;
  public projectManagerId?: number;
  public estimatedTotalHours?: number;
  public actualTotalHours?: number;
  public estimatedLaborCost?: number;
  public actualLaborCost?: number;
  public estimatedMaterialCost?: number;
  public actualMaterialCost?: number;
  public projectStatusId: number;

  constructor(projectSn: string,
    projectName: string,
    projectDescription: string,
    clientId: number,
    divisionId: number,
    estimatedStartDate: Date,
    estimatedEndDate: Date,
    actualStartDate: Date,
    actualEndDate: Date,
    projectManagerId: number,
    estimatedTotalHours: number,
    actualTotalHours: number,
    estimatedLaborCost: number,
    actualLaborCost: number,
    estimatedMaterialCost: number,
    actualMaterialCost: number,
    projectStatusId: number,) {

    this.projectSn = projectSn;
    this.projectName = projectName;
    this.projectDescription = projectDescription;
    this.clientId = clientId;
    this.divisionId = divisionId;
    this.estimatedStartDate = estimatedStartDate;
    this.estimatedEndDate = estimatedEndDate;
    this.actualStartDate = actualStartDate;
    this.actualEndDate = actualEndDate;
    this.projectManagerId = projectManagerId;
    this.estimatedTotalHours = estimatedTotalHours;
    this.actualTotalHours = actualTotalHours;
    this.estimatedLaborCost = estimatedLaborCost;
    this.actualLaborCost = actualLaborCost;
    this.estimatedMaterialCost = estimatedMaterialCost;
    this.actualMaterialCost = actualMaterialCost;
    this.projectStatusId = projectStatusId;
  }


}
